File: backend/ms_push/src/main/java/wevioo/tn/ms_push/dtos/request/SchedulePushRequest.java
package wevioo.tn.ms_push.dtos.request;

import lombok.Data;
import wevioo.tn.ms_push.entities.WebPushSubscription;

import java.time.LocalDateTime;
import java.time.ZoneId;
import java.util.Map;
import java.util.Set;

@Data
public class SchedulePushRequest {
    private Set<WebPushSubscription> webPushSubscriptions;
    private Long templateId;
    private String name;
    private Map<String,String> placeHolders;
    private LocalDateTime dateTime;
    private ZoneId timeZone;
    private Long userId;
}

File: frontend/src/models/push/SchedulePushRequest.ts
import { WebPushSubscription } from "./WebPushSubscription";

export interface SchedulePushRequest {
    templateId:number;
    webPushSubscriptions:WebPushSubscription[];
    placeholders: { [key: string]: string };
    userId:number;
    timeZone:string;
    dateTime:string;
    name:string
    isAdmin:boolean
}